### ISSUE TITLE
analysis server doesn't seem to scale well with a larger number of files

### ISSUE THREAD

**Issue by pq** [dart-lang/sdk#25130]
_From @Andersmholmgren on November 7, 2015 23:46_

I am using intellij and reached a point where things became unusable due to delays between typing something and the analyser catching up and giving me feedback.

I reduced the number of dart projects that I was editing concurrently within the one intellij project and it alleviated the problem somewhat.

This is however, very annoying as it is very convenient to be able to edit these projects at once.

_Copied from original issue: dart-lang/analyzer_cli#78_

**Comment by pq** [comment]
_From @zoechi on November 8, 2015 13:42_

I saw mentioned recently that DartAnalyzer only analyzes or somehow prioritizes files that are opened in the IDE and shown to the user. I maintain [a project with small and simple Polymer examples](https://github.com/bwu-dart-playground/polymer1), each in it's own Dart package with it's own `pubspec.yaml`.
I wonder if this is really the case, because working on only one of the examples for a while often bogs down WebStorm and after restarting it takes quite a while until the progress bar showing "analyzing ..." is completed and WebStorm becomes responsible again for a while. Analyzer seems to do quite a lot of work for 2 or 3 opened Dart files.
This seems to be less of an issue in much bigger single package projects.

I wasn't yet able to figure out a consistent pattern when WebStorm becomes unresponsive. It seems often to be related to opening/closing other Dart projects in other WebStorm windows. 
I saw that a bug related to this was fixed recently, not sure if this is already active in my Dart version (recent bleeding edge)
